v0/destinations/userlist/transform.js

```diff
@@ -0,0 +1,52 @@
+const _ = require("lodash");
+
+const { EventType } = require("../../../constants");
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig
+} = require("../../util");
+
+const { endpoint } = require("./config");
+
+function isBlank(value) {
+  return _.isEmpty(_.toString(value));
+}
+
+function buildResponse(message, destination) {
+  const response = defaultRequestConfig();
+  const { pushKey } = destination.Config;
+
+  response.endpoint = endpoint;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json",
+    Authorization: `Push ${pushKey}`
+  };
+  response.body.JSON = message;
+  response.statusCode = 200;
+
+  return response;
+}
+
+function processSingleMessage(message, destination) {
+  const messageType = message.type.toLowerCase();
```

Comment:
```suggestion
  const messageType = message.type && message.type.toLowerCase();
```

Reply:
I'd expect that the existence of required fields (according to the [spec](https://docs.rudderstack.com/rudderstack-api-spec#common-fields)) like this would be checked _before_ the message gets passed to the transforms. Otherwise what's the point of having a spec? 😄 

Anyways, fixed this.

Reply:
I agree with you. We don't have a top-level validator for the events, and at some times, we saw that some of the events were messed up because of a bad transformation function. This is for fail-safety. We're working on the top level validator. Then we'll clean-up these sections of the code.